proc/process.c

```diff
@@ -578,13 +585,26 @@ int process_load(process_t *process, vm_object_t *o, offs_t base, size_t size, v
 		reloffs = 0;
 		prot = PROT_USER;
 		flags = MAP_NONE;
+		paddr = (char *)ehdr + phdr->p_offset;
 
 		if (phdr->p_flags & PF_R)
 			prot |= PROT_READ;
 
-		if (phdr->p_flags & PF_X)
+		if (phdr->p_flags & PF_X) {
 			prot |= PROT_EXEC;
 
+			if (process->imapp != NULL &&
+					(((ptr_t)base < (ptr_t)process->imapp->start) ||
+					((ptr_t)base > (ptr_t)process->imapp->stop))) {
```

Comment:
You don't check whether the size of the elf fits in the map boundaries.

Reply:
I don't really have to, as `vm_mmap` below will verify that for me